### PR TITLE
WiP: Add index.mjs export (typescript to ES2020)

### PR DESCRIPTION
**Pull request by 1000i100**
Implement: #16

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "strict": true,
     "target": "es2020",
     "module": "commonjs",
+    "moduleResolution": "node",
```
Comment (paulmillr) [inline tsconfig.json:6]: what's that?

Comment (1000i100) [reply tsconfig.json:6]: It what i found to make `tsc -m ES2020` succeed.
I'm very unfamiliar with TypeScript, so i don't know what it exactly do.
I've just look in the documentation : https://www.typescriptlang.org/docs/handbook/module-resolution.html#module-resolution-strategies so if module is commonjs moduleResolution default is node, else it's classic. By adding this line i force node resolution so build pass.

**Comment by 1000i100** [comment]
I've tagged the merge-request in WiP because in mjs it's converted to node but fail to run in node.js environnement because mjs files don't allow require().

I've ask for help here : https://stackoverflow.com/questions/66936773/isomorphic-code-for-native-node-js-and-browser-crypto

I've just look in tweetnacl why it work in my case and not noble-ed25519 and found : 
tweetnacl only use require('crypto') / window.crypto for random generation.
noble-ed25519 use it for random and for sha512.

What i currently use need sha512 but not random generation, so tweetnacl work and not noble-ed25519 (it fail in node because it's imported in an mjs file unable to use require() .
But your implementation is probably faster than tweetnacl.
So yes, let's wait for node 16 or good help there : https://stackoverflow.com/questions/66936773/isomorphic-code-for-native-node-js-and-browser-crypto

**Comment by paulmillr** [comment]
I think we have ESM support now

**Pull request closed without merging**